pkg/operator/controller/gatewayapi/status.go

```diff
@@ -0,0 +1,80 @@
+package gatewayapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	operatorcontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller/status"
+)
+
+// setUnmanagedGatewayAPICRDNamesStatus sets the status of the ingress cluster operator
+// with the names of the unmanaged Gateway CRDs.
+func (r *reconciler) setUnmanagedGatewayAPICRDNamesStatus(ctx context.Context, crdNames []string) error {
+	return r.setClusterOperatorStatusExtension(ctx, &status.IngressOperatorStatusExtension{
+		UnmanagedGatewayAPICRDNames: strings.Join(crdNames, ","),
+	})
+}
+
+// setClusterOperatorStatusExtension attempts to ensure that the ingress cluster operator's status
+// is updated with the given extension. Returns an error if failed to update the status.
+func (r *reconciler) setClusterOperatorStatusExtension(ctx context.Context, desiredExtension *status.IngressOperatorStatusExtension) error {
+	have, current, err := r.currentClusterOperator(ctx, operatorcontroller.IngressClusterOperatorName())
+	if err != nil {
+		return err
+	}
+	if !have {
+		return fmt.Errorf("cluster operator %q not found", operatorcontroller.IngressClusterOperatorName().Name)
```

Comment:
```suggestion
		return fmt.Errorf("ingress operator %q not found", operatorcontroller.IngressClusterOperatorName().Name)
```

Reply:
My code gives the following error message:
```
cluster operator "ingress" not found
```

What you are proposing would give the following message:
```
ingress operator "ingress" not found
```
which looks redundant.